### PR TITLE
fix(tree): make findNode/getParentNode null-tolerant

### LINKED ISSUE #2424: bug(tree): findNode/getParentNode throw on null treeData
## Description

Tree store actions (`deleteAiDataCollection`, `deleteAiDataCollectionElement`, `updateTreeNodeDataByKey`, `closeConnection`, `updateOriginalTitleByNodeId`, `getChildrenByNodeId`) pass `get().treeData!` into `findNode`/`getParentNode`. `treeData` is legitimately `null` (initial value, `getTreeData`, `clearTreeStore` null it). `findNode`/`getParentNode` dereference `tree.length` with no null guard → `null.length` NPE. The `!` non-null assertions at call sites mask the nullable type.

## Location

`chat2db-community-client/src/utils/index.ts:384-397` (getParentNode), `:425-438` (findNode); call sites `src/store/tree/index.tsx:501,526,568,655,669,678`.

## Impact

Rename/delete/close-connection actions NPE when `treeData` is null (e.g. right after login or during a refresh).

## Suggested fix

Make `findNode`/`getParentNode` null-tolerant (guard `if (!tree || !tree.length) return undefined;` and drop the `!` on the return), so a null tree yields `undefined` instead of throwing. Single-point fix covering all call sites.

## Related existing

None.

### PR DESCRIPTION
**Pull request by Aias00**
## What

Tree store actions can run while `treeData` is legitimately `null` during initialization, refresh, or after clearing the store. The previous lookup helpers dereferenced the tree and also claimed a non-null return type even when a node was not found.

## Fix

- Move the recursive lookup logic into `treeNodeLookup.ts`.
- Accept nullable tree input and return `TreeNodeData | undefined` from `findNode` and `getParentNode`.
- Guard all affected rename, delete, close-connection, and refresh call sites with explicit fallback behavior.
- Refresh the full tree when a parent node cannot be located.
- Avoid `splice(-1, 1)` when the target datasource is absent.
- Add focused lookup tests for null trees, missing keys, nested matches, and parent lookup.

## Verification

- `yarn test:tree-node-lookup`
- `yarn test:tree-data-update`
- ESLint on all changed TypeScript files with zero warnings
- `yarn build:web:community`

Fixes #2424

### CODE REVIEW

File: chat2db-community-client/src/utils/index.ts [hunk 1]
```diff
@@ -424,7 +427,10 @@ export function removeSubkeys(expandedKeys: React.Key[], tree: TreeNodeData[], t
 // Find the node of the tree
 export const findNode = (key: React.Key, tree: TreeNodeData[]): TreeNodeData => {
   let finalNode: any;
-  for (let i = 0; i < tree.length; i++) {
+  // treeData is legitimately null (initial, getTreeData, clearTreeStore) and call
+  // sites pass it with `!`; guard the loop so a null tree yields undefined instead
+  // of throwing null.length.
```
[the remark this answers is not in the document]
Comment (openai0229) [reply chat2db-community-client/src/utils/index.ts]: Addressed in 54dddf895: `findNode` now exposes its real nullable contract, and all changed call sites handle `undefined` safely instead of relying on non-null assertions. The new lookup tests cover null input and missing keys.

File: chat2db-community-client/src/utils/index.ts [hunk 2]
```diff
@@ -383,7 +383,10 @@ export function getLanguageType(language?: string) {
 // Find the parent node of a tree node
 export const getParentNode = (key: React.Key, tree: TreeNodeData[]): TreeNodeData => {
   let finalNode: any;
-  for (let i = 0; i < tree.length; i++) {
+  // treeData is legitimately null (initial, getTreeData, clearTreeStore) and call
+  // sites pass it with `!`; guard the loop so a null tree yields undefined instead
+  // of throwing null.length.
```
[the remark this answers is not in the document]
Comment (openai0229) [reply chat2db-community-client/src/utils/index.ts]: Addressed in 54dddf895: the lookup helpers now accept nullable tree input, return `TreeNodeData | undefined`, and affected callers explicitly guard or fall back before using the result. Focused null and not-found tests were added.

**Review by openai0229 — APPROVED** [review]
Reviewed after the nullable-contract and call-site fixes. The focused lookup tests, affected-file ESLint, and full Community frontend build pass locally.
